File: app/src/main/java/lajevski/radoslav/tesonetparty/di/ActivityComponent.java
package lajevski.radoslav.tesonetparty.di;

import dagger.Component;
import lajevski.radoslav.tesonetparty.ui.activities.LoginActivity;
import lajevski.radoslav.tesonetparty.ui.fragments.servers.ServersFragment;
import lajevski.radoslav.tesonetparty.ui.fragments.login.LoginFragment;

/**
 * Created by Radoslav on 1/16/2018.
 */

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(LoginActivity activity);

    void inject(LoginFragment fragment);

    void inject(ServersFragment fragment);
}

File: app/src/main/java/com/example/ievazygaite/androidparty/di/component/ActivityComponent.kt
package com.example.ievazygaite.androidparty.di.component

import com.example.ievazygaite.androidparty.di.module.ActivityModule
import com.example.ievazygaite.androidparty.ui.main.MainActivity
import dagger.Component

@Component(modules = [ActivityModule::class])
interface ActivityComponent {
    fun inject(mainActivity: MainActivity)
}